File: 2022.01.22/python/2750.py
n = int(input())
nums = []
for _ in range(n):
  nums.append(int(input()))

def insertion_sort(arr):
  for i in range(1,len(arr)):
    key_num = arr[i]
    idx = i-1
    for j in range(i-1,-1,-1):
      if key_num<arr[j]:
        arr[j+1] = arr[j]
        idx -= 1
      else:
        break
    if idx != i-1 : arr[idx+1] = key_num
insertion_sort(nums);
for num in nums:
  print(num)

File: Boj.단계별/12_정렬/javascript/2750.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const nums = input.slice(1,input.length-1).reduce((acc,cur) => {acc.push(Number(cur)); return acc},[]);

const insertionSort = (arr) => {
  let len = arr.length;
  let keyNum;
  let cnt;
  for (let i = 1 ; i<len ; i++){
    keyNum = arr[i];
    idx = i-1;
    for (let j = i-1 ; j>=0; j--){
      if (keyNum<arr[j]) {
        arr[j+1] = arr[j];
        idx--;
      }else {
        break;
      }
    if (idx!== i-1) arr[idx+1] = keyNum;
    }
  }
  return arr;
}
insertionSort(nums);
nums.reduce((prev,cur) => console.log(cur),0)
